django/src/rdwatch/tasks/__init__.py

```diff
@@ -249,27 +213,23 @@ def get_siteobservations_images(
             cloudcover = capture.cloudcover
             count += 1
             output = f'tile_image_{observation.siteeval}_nonobs_{count}.jpg'
-            with open(output, 'wb') as f:
-                f.write(bytes)
-            with open(output, 'rb') as imageFile:
-                image = File(imageFile, output)
-                found = SiteImage.objects.filter(
+            image = File(io.BytesIO(bytes), name=output)
+            found = SiteImage.objects.filter(
+                siteeval=baseSiteEval,
+                timestamp=capture.timestamp,
+                source=baseConstellation,
+            )
+            if found.count() > 0:
```

Comment:
```suggestion
            if found.exists():
```
If we just want to check for existence, this should generate a more efficient query than an expensive `COUNT`. https://docs.djangoproject.com/en/4.2/ref/models/querysets/#django.db.models.query.QuerySet.exists